### PR TITLE
ArgumentParser, MyPy, Flake8, & Circular Import Fixes

### PR DESCRIPTION
**Pull request by ric-evans**
Module now passes with `pycycle --here && pytest --mypy --flake8`.

### CODE REVIEW

**Review by dsschult — APPROVED** [review]
Other than the two failing tests with `super()` this looks good.

**Comment by ric-evans** [comment]
> Other than the two failing tests with `super()` this looks good.

Those two tests ended up being a rabbit hole of fixes.

**Review by dsschult — APPROVED** [review]

File: rest_tools/server/arghandler.py [hunk 1]
```diff
@@ -18,11 +17,9 @@ class _NoDefaultValue:  # pylint: disable=R0903
 NO_DEFAULT = _NoDefaultValue()
 
 
-def _get_json_body_arguments(
-    request_handler: tornado.web.RequestHandler,
-) -> Dict[str, Any]:
+def _parse_json_body_arguments(request_body: bytes,) -> Dict[str, Any]:
```
Comment (blinkdog) [inline rest_tools/server/arghandler.py:20]: The comma after `bytes` here looks weird to my eyes when not followed by another parameter.

Comment (ric-evans) [reply rest_tools/server/arghandler.py]: yup, the black formatter does that sometimes when multi-liners collapse to one-liners

File: rest_tools/server/config.py [hunk 2]
```diff
@@ -4,14 +4,25 @@
 import os
 from typing import cast, Dict, Mapping, Optional, Sequence, Union
 
-OptionalDict = Mapping[str, Optional[str]]
-KeySpec = Union[str, Sequence[str], OptionalDict]
 RetVal = Union[str, int, float, bool]
+OptionalDict = Mapping[str, Optional[RetVal]]
+KeySpec = Union[str, Sequence[str], OptionalDict]
```
Comment (blinkdog) [inline rest_tools/server/config.py:9]: My natural desire to alphabetize wants these in a different order, but that's not a problem with the code, only me. :smile:

**Review by blinkdog — APPROVED** [review]
LGTM

**Pull request merged**